packages/selenium/src/AxeBuilder.cs

```diff
@@ -254,75 +256,157 @@ private AxeResult AnalyzeRawContext(object rawContextArg)
+            Stack<object> frameStack
+        )
+        {
+            var serializedContext = JsonConvert.SerializeObject(context.Context, AxeJsonSerializerSettings.Default);
+            var windowHandle = _webDriver.CurrentWindowHandle;
+
+            var partialResults = new List<object>();
+
+            // get the proper selector the frame and switch to it
+            var selectorForCurrentFrame = _webDriver.ExecuteScript(EmbeddedResourceProvider.ReadEmbeddedFile("shadowSelect.js"), JsonConvert.SerializeObject(context.Selector, AxeJsonSerializerSettings.Default));
+            _webDriver.SwitchTo().Frame(selectorForCurrentFrame as IWebElement);
+
+
+            try
+            { // finally: pop frameStack
+                frameStack.Push(context.Selector);
+
+                // inject axe and configure it
+                ConfigureAxe();
+
+                try
+                {
+                    string frameResultString = (string)_webDriver.ExecuteAsyncScript(EmbeddedResourceProvider.ReadEmbeddedFile("runPartial.js"), serializedContext, options);
+                    // Important to deserialize because we want to reserialize as an
+                    // array of object, not an array of strings.
+                    partialResults.Add(JsonConvert.DeserializeObject<object>(frameResultString));
+                }
+                catch (Exception e)
                 {
+                    Trace.TraceWarning($"Error executing runPartial. Error message: {e.ToString()}");
                     _webDriver.SwitchTo().ParentFrame();
+
+                    partialResults.Add(null);
+
+                    return partialResults;
                 }
-            }
 
+                var frameContexts = GetFrameContexts(serializedContext);
 
+                frameContexts.ForEach(fContext =>
+                {
+                    try
+                    {
+                        partialResults.AddRange(RunPartialRecursive(
+                            fContext,
+                            options,
+                            frameStack
+                        ));
+                    }
+                    catch (WebDriverTimeoutException)
+                    {
+                        Trace.TraceWarning("RunPartial for sub-frame threw. Switching to parent.");
+                        _webDriver.SwitchTo().Window(windowHandle);
+                        foreach (var frameSelector in frameStack)
+                        {
+                            var selector = _webDriver.ExecuteScript(EmbeddedResourceProvider.ReadEmbeddedFile("shadowSelect.js"), JsonConvert.SerializeObject(frameSelector, AxeJsonSerializerSettings.Default));
+                            if (selector is IWebElement el)
+                            {
+                                _webDriver.SwitchTo().Frame(el);
+                            }
+                        }
+                        partialResults.Add(null);
+                        return;
```

Comment:
This should return `partialResults` otherwise you add nothing from the subframe to the final result. Something like this tells me that this code is not doing what we expect. Otherwise I would expect the frame results to be empty due to this return

Reply:
Looking at the lazy-loaded iframe test, I can see that it only adds the lazy loaded iframe at the top level and not nested within an iframe. That is probably why this code after 2 rewrites works even though it would have problems if the lazy loaded iframe was nested within another frame. 

I'll look into updating the test fixture in order to flag some of these things.